### PR TITLE
Implement getUserDetails endpoint

### PR DESCRIPTION
**Pull request by soumitradev**

### CODE REVIEW

File: src/controllers/user/getUserDetails.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        message: "id must be a non-empty string",
+      })
+      .uuid({ message: "id must be a valid uuid" }),
+  }),
+  // auth temp replacement
+  query: z.object({
+    authEmail: z
+      .string({
+        invalid_type_error: "authEmail not a string",
+        required_error: "authEmail is a required path parameter",
```
Comment (majimearun) [inline src/controllers/user/getUserDetails.ts:24]: error message says path param?

File: src/controllers/user/getUserDetails.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    let user: User | null = null;
+
+    try {
+      user = await userRepository
+        .createQueryBuilder("user")
+        .select()
+        .leftJoinAndSelect(
+          "user.timetables",
+          "timetable",
+          "(user.email <> :email and timetable.private = :private and timetable.draft = :draft) or (user.email = :email)",
```
Comment (majimearun) [inline src/controllers/user/getUserDetails.ts:53]: Are you sure you want to return all timetable details? Wouldn't the id, relevant metadata  and name suffice?

You could get the timetable itsef from the ID on subsequent request for a particular timetable.

Each empty timetable itself is over 250B, so the response size would get pretty big very fast.

**Review by majimearun — CHANGES_REQUESTED** [review]
Except for the few nitpicky things (which might not be a problem, just mentioned it jic), looks LGTM

File: src/controllers/user/getUserDetails.ts (hunk 2 above)
Comment (soumitradev) [reply src/controllers/user/getUserDetails.ts:52]: Resolved in 2f30906

**Review by majimearun — APPROVED** [review]

**Pull request merged**